lib/spack/spack/ci.py

```diff
@@ -2176,13 +2176,13 @@ def __init__(self, ci_cdash):
     def args(self):
         return [
             "--cdash-upload-url",
-            self.upload_url,
+            f'"{self.upload_url}"',
             "--cdash-build",
-            self.build_name,
+            f'"{self.build_name}"',
```

Comment:
This extra double quoting seems to be broken on windows 😞 What ends up in the generated `install.sh` is a bunch of commands with `""asdfasdfasdf ()""` which is incorrect.

I know we did this for windows do handle things not being quoted by default. We may just need to have a `quote_on_windows` function here.